File: engines/py/anza/models/envelope.py
from dataclasses import dataclass
from typing import Any, Optional


@dataclass
class Envelope:
    slot: str
    ts: int
    html: str
    sig: Optional[str] = None
    css: Optional[str] = None

    def message(self) -> str:
        return f"{self.ts}:{self.slot}:{self.html}"

    def to_dict(self) -> dict[str, Any]:
        data: dict[str, Any] = {
            "slot": self.slot,
            "ts": self.ts,
            "html": self.html,
        }
        if self.sig is not None:
            data["sig"] = self.sig
        if self.css is not None:
            data["css"] = self.css
        return data

File: engines/rust/src/models/envelope.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct Envelope {
  pub slot: String,
  pub ts: u64,
  pub html: String,
  #[serde(skip_serializing_if = "Option::is_none")]
  pub sig: Option<String>,
  #[serde(skip_serializing_if = "Option::is_none")]
  pub css: Option<String>,
}

impl Envelope {
  pub fn new<S: Into<String>>(slot: S, ts: u64, html: S) -> Self {
    Self {
      slot: slot.into(),
      ts,
      html: html.into(),
      sig: None,
      css: None,
    }
  }

  pub fn sign<S: Into<String>>(mut self, signature: S) -> Self {
    self.sig = Some(signature.into());
    self
  }

  pub fn style<S: Into<String>>(mut self, css: S) -> Self {
    self.css = Some(css.into());
    self
  }

  pub fn message(&self) -> String {
    format!("{}:{}:{}", self.ts, self.slot, self.html)
  }
}
